File: app/src/main/java/com/raaveinm/ramp/MainActivity.java
package com.raaveinm.ramp;

import static com.raaveinm.ramp.ext.ExtMethods.getRandom;

import android.annotation.SuppressLint;
import android.content.res.AssetFileDescriptor;
import android.media.AudioAttributes;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;

import android.media.MediaMetadataRetriever;
import android.view.View;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.SeekBar;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.recyclerview.widget.RecyclerView;

import android.media.MediaPlayer;

import com.raaveinm.ramp.ext.ExtMethods;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;

/**
 * List of questions
 * 1. How to export value of playPauseStatus to another lifeCycle? (solved)
 * 2. How to Import media from local storage?
 * 3. How to Create media query?
 * 4. How to optimize usage of resources?
 * 5. Why did it throws error, if declaration was made in commented section? (solved)
 * 6. How to create app logo?
 * 7. How to make smooth transition in two photos or smooth disappearing of button (for example)?
 * 8. RecyclerView     com.raaveinm.ramp       E  No adapter attached; skipping layout
 * 9. How to get Track Info?
 */

public class MainActivity extends AppCompatActivity {
    String TAG = "MainActivity";
    MediaPlayer player = new MediaPlayer();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        PlayerFragment.newInstance("null","null");
    }


    @SuppressLint("PrivateResource")
    protected void onStart(){
        super.onStart();
    }

    protected void onPause(){
        super.onPause();
    }

    protected void onResume(){
        super.onResume();
    }

    @Override
    protected void onRestart() {
        super.onRestart();
    }

    protected void onStop() {
        super.onStop();
    }

    protected void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/com/raaveinm/ramp/MainActivity.kt
package com.raaveinm.ramp

import android.Manifest
import android.Manifest.permission.READ_EXTERNAL_STORAGE
import android.Manifest.permission.READ_MEDIA_AUDIO
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.util.Log
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.core.content.ContextCompat
import com.raaveinm.ramp.ui.layouts.PlayerScreen
import com.raaveinm.ramp.ui.theme.RampTheme



class MainActivity : ComponentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        var requirePermission = ""

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            requirePermission = READ_MEDIA_AUDIO
        } else {
            requirePermission = READ_EXTERNAL_STORAGE
        }

        setContent {
            RampTheme {
                requestPermission(requirePermission)
                val context = applicationContext

                PlayerScreen()
            }
        }
    }

    fun requestPermission(type: String):Boolean {
        when (type) {
            READ_MEDIA_AUDIO ->{
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
                    if (ContextCompat.checkSelfPermission( this, READ_MEDIA_AUDIO
                        ) == PackageManager.PERMISSION_GRANTED
                    )
                    return true
                }else{
                    Log.e("MainActivity", "Requested unsupported permission")
                    return false
                }
            }
            READ_EXTERNAL_STORAGE -> {
                if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
                    if (ContextCompat.checkSelfPermission( this, READ_EXTERNAL_STORAGE
                        ) == PackageManager.PERMISSION_GRANTED
                    )
                    return true
                }else{
                    Log.e("MainActivity", "Requested unsupported permission")
                    return false
                }
            }
        }
        return false
    }
}